locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsSshMachineLocation.java

```diff
@@ -162,8 +162,12 @@ public void init() {
     @Override
     public void rebind() {
         super.rebind();
-        ComputeServiceContext context = jcloudsParent.getComputeService().getContext();
-        runScriptFactory = context.utils().injector().getInstance(RunScriptOnNode.Factory.class);
+        
+        if (jcloudsParent != null) {
+            // can be null on rebind, if location has been "orphaned" somehow
+            ComputeServiceContext context = jcloudsParent.getComputeService().getContext();
+            runScriptFactory = context.utils().injector().getInstance(RunScriptOnNode.Factory.class);
+        }
```

Comment:
The problem is the bug happened a long time ago, to persist these locations (or more likely not delete them when the parent was deleted). I'm hoping that's fixed but it might not be.